pkgs/development/compilers/corretto/mk-corretto.nix

```diff
@@ -21,14 +27,34 @@ let
   # The version scheme is different between OpenJDK & Corretto.
   # See https://github.com/corretto/corretto-17/blob/release-17.0.8.8.1/build.gradle#L40
   # "major.minor.security.build.revision"
+  #
+  appleFrameworks = darwin.apple_sdk_11_0.frameworks;
+  metalFrameworks = [
+    appleFrameworks.Accelerate
+    appleFrameworks.Metal
+    appleFrameworks.MetalKit
+    appleFrameworks.MetalPerformanceShaders
+  ];
+
 in
 jdk.overrideAttrs (finalAttrs: oldAttrs: {
   inherit pname version src;
   name = "${pname}-${version}";
 
-  nativeBuildInputs = oldAttrs.nativeBuildInputs ++ [ jdk gradle rsync ];
+  nativeBuildInputs = oldAttrs.nativeBuildInputs ++ [ jdk gradle rsync ] ++ lib.optionals stdenv.isDarwin
+    (
+    [ coreutils zip autoconf which xcbuild darwin.xattr darwin.stubs.setfile ]
+      ++
+      metalFrameworks
+    );
+
+  configurePhase = ''
+    runHook preConfigure
+
+    bash configure --with-xcode-path=$(xcode-select -p)
```

Comment:
@pcasaretto I think we should rather stick to `dontConfigure = true;` due to the issue you [mentioned](https://github.com/NixOS/nixpkgs/pull/290808#issuecomment-1977033010). If you need to provide the xcode path, you could use the approach that I have already used [here](https://github.com/rollf/nixpkgs/blob/ad70200fc52e220150f7ec6739537bf6cfa832b0/pkgs/development/compilers/corretto/11.nix#L33) (i.e. simply adapt the gradle file to add the necessary options)

Reply:
Maybe (guessing here):

```bash
substituteInPlace $file --replace "command += project.correttoCommonFlags" "command += project.correttoCommonFlags + ['--with-xcode-path=$(xcode-select -p)']"
```

Reply:
Agreed. In my current local iteration I'm using something similar.

```
      export EXTRA_CORRETTO_CONFIG_OPTIONS="--with-xcode-path=$(xcode-select -p) --disable-precompiled-headers"
      gradle -Pcorretto.extra_config="$EXTRA_CORRETTO_CONFIG_OPTIONS" --console=plain --no-daemon ${task}
```